### PR TITLE
Github workflows and unit tests

### PR DESCRIPTION
**Pull request by DOCtorActoAntohich**
What's done:
  - improved code
  - added linter
  - added stupid unit tests
  - added workflows to test code + build and push docker image

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -2,29 +2,23 @@ FROM python:3.10
 WORKDIR /app
 
 
+EXPOSE 8000
```
Comment (Andrew174194) [inline docker/Dockerfile:5]: Thanks for EXPOSE 🙏

File: tests/test_time_use_case.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    krasnoyarsk_time = krasnoyarsk_use_case.execute()
+
+    current_moscow_time = datetime.datetime.now(tz=pytz.timezone("Europe/Moscow"))
+
+    moscow_dt = current_moscow_time - moscow_time
+    krasnoyarsk_moscow_dt = krasnoyarsk_time - moscow_time
+    assert moscow_dt.seconds < 1, "Invalid time in Moscow timezone"
+    assert krasnoyarsk_moscow_dt.seconds < 1, (
+        "Time delta between Moscow and Krasnoyarsk should be "
+        "negligibly small because timezones are ignored"
```
Comment (ntdesmond) [inline tests/test_time_use_case.py:24]: i hate datetime timezones i hate datetime timezones i hate datetime timezones i hate datetime timezones i hate datetime timezones i hate datetime timezones i hate datetime timezones i hate datetime timezones i hate datetime timezones i hate datetime timezones

**Review by ntdesmond — APPROVED** [review]
e

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+name: Build latest
+
+on:
+  pull_request:
+    branches:
+      - "master"
+      - "dev"
```
[the remark this answers is not in the document]
Comment (smthngslv) [reply .github/workflows/build.yaml:7]: push on MR is not necessary, i guess

File: .github/workflows/checks.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    - uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip setuptools wheel
+        python -m pip install -r requirements-dev.txt
+        python -m pip install flake8
```
Comment (smthngslv) [inline .github/workflows/checks.yaml:22]: why flake8 is not in requirements

File: .github/workflows/checks.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    - name: Use Linter
+      run: |
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
```
Comment (smthngslv) [inline .github/workflows/checks.yaml:25]: you can move all params into flake8 config, I guess

File: .github/workflows/build.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (DOCtorActoAntohich) [reply .github/workflows/build.yaml:7]: I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS
I HATE GITHUB WORKFLOWS

File: .github/workflows/checks.yaml (hunk 4 above)
Comment (DOCtorActoAntohich) [reply .github/workflows/checks.yaml:22]: because it sucks

File: docker/Dockerfile (hunk 1 above)
Comment (DOCtorActoAntohich) [reply docker/Dockerfile:5]: it has no impact + compose is way better...

but still, вы являетесь добро пожаловать

**Pull request merged**